### PR TITLE
Task/22627 Add sales type for stories

### PR DESCRIPTION
**Pull request by tnminh00**
==> Title: refs task/22627 add sales type for stories

==> Redmine ticket: https://edu-redmine.sun-asterisk.vn/issues/22627

==> Attachment:

### CODE REVIEW

**Comment by tnminh00** [comment]
ready

File: app/models/story.rb [hunk 1]
```diff
@@ -8,9 +8,12 @@ class Story < ApplicationRecord
   has_many :comments, as: :commentable
   has_many :follows, dependent: :destroy
   has_many :users, through: :follows
+  has_many :payments
+  has_many :buyers, through: :payments, source: :user
   has_one :rate_average, -> { where dimension: "rating"}, as: :cacheable, class_name: RatingCache.name, dependent: :destroy
   ratyrate_rateable "rating"
 
+  enum sales_type: [:free, :sale]
```
Comment (ngocvt-0484) [inline app/models/story.rb:16]: enum e khai báo dưới dạng hash đi

File: db/seeds.rb [hunk 2]
```diff
@@ -26,3 +27,15 @@
   s.category_stories.create category: sample.first
   s.category_stories.create category: sample.second
 end
+
+Story.where("id < 20").each do |s|
+  s.update_attributes sales_type: 1
```
Comment (ngocvt-0484) [inline db/seeds.rb:32]: 1 thay bằng enum đi e

**Comment by tnminh00** [comment]
ready

**Pull request merged**